### ISSUE TITLE
possible Service $name collision

### ISSUE THREAD

**Issue by neoice** [elasticsearch/puppet-elasticsearch#181]
https://github.com/elasticsearch/puppet-elasticsearch/blob/master/manifests/service/init.pp#L178

the `elasticsearch::instance` `$name` gets passed into this `Service` declaration.

code like this:

```
elasticsearch::instance {'logstash: }
class { '::logstash': }
```

will create two different declarations of `Service['logstash']`, resulting in a "duplicate resource" error.

a simple fix would be to namespace the `Service` name, maybe something like `"elasticsearch-instance-${name}"`.

<sup>I haven't tested my "minimal test case", but it's extracted from a much larger manifest that I'm not able to share in full. I'd be willing to cook up a full manifest example if needed.</sup>

**Comment by electrical** [comment]
Hi,

Sorry for all the delays.
In the last couple days i've been working on all the issues and PR's and other improvements.
I'm currently working on a PR to solve this specific issue by renaming all the `$name` vars for the service to `elasticsearch-instance-$name` which should solve this issue.

**Comment by electrical** [comment]
PR https://github.com/elasticsearch/puppet-elasticsearch/pull/240 should solve this issue.

**Comment by electrical** [comment]
This has been solved with merging #240
